pandas/tests/reshape/test_crosstab.py

```diff
@@ -795,6 +795,33 @@ def test_margin_normalize_multiple_columns(self):
         expected.index.name = "C"
         tm.assert_frame_equal(result, expected)
 
+    def test_margin_support_Float(self):
+        # GH 50313
+        # use Float64 formats and function aggfunc with margins
+        df = DataFrame(
+            {"A": [1, 2, 2, 1], "B": [3, 3, 4, 5], "C": [-1.0, 10.0, 1.0, 10.0]}
+        )
+        df = df.astype({"C": "Float64"})
+        result = crosstab(
+            df["A"],
+            df["B"],
+            values=df["C"],
+            aggfunc="sum",
+            margins=True,
+        )
+        expected = DataFrame(
+            [
+                [-1.0, pd.NA, 10.0, 9.0],
+                [10.0, 1.0, pd.NA, 11.0],
+                [9.0, 1.0, 10.0, 20.0],
+            ],
+            index=[1, 2, "All"],
+            dtype="Float64",
+        )
+        expected.columns = Index([3, 4, 5, "All"], dtype="object", name="B")
```

Comment:
Same here, please define expected in one go

Reply:
Thank you, @phofl, for your comments. I updated my PR as you suggested.